### PR TITLE
PHP composer updates + npm

### PR DESCRIPTION
**Pull request by melroy89**
- Update doctrine packages
- Install missing ux-twig-component recipe (introducing `twig_component.yaml` config)
- Upgrade doctrine/annotations to v2.0 (major version upgrade) + annotation fixes
- Combine the two nodeinfo paths to a single path
- Update Symfony packages + other minor package updates
- Upgrade PHPUnit to v10 + fix deprecation
- Ran phpunit xml migration script which update [the xml.dist file which upgrade to v10](https://docs.phpunit.de/en/10.4/configuration.html) as well
   - `vendor/bin/phpunit --migrate-configuration`
- Npm update

### CODE REVIEW

File: config/kbin_routes/activity_pub.yaml [hunk 1]
```diff
@@ -1,144 +1,140 @@
...
-  path: '/nodeinfo/2.1.{_format}'
-  methods: [ GET ]
-  format: json
+    controller: App\Controller\ActivityPub\NodeInfoController::nodeInfo
+    path: '/.well-known/nodeinfo'
+    methods: [GET]
+
+ap_node_info_v2:
+    controller: App\Controller\ActivityPub\NodeInfoController::nodeInfoV2
+    path: '/nodeinfo/{version}.{_format}'
```
Comment (melroy89) [inline config/kbin_routes/activity_pub.yaml:18]: Noe thsi line.. We combine the two nodeinfo paths to a single config, allowing to have 2.0 or 2.1 as `{version}`

File: bin/phpunit [hunk 2]
```diff
@@ -8,7 +8,7 @@ if (!ini_get('date.timezone')) {
 if (is_file(dirname(__DIR__).'/vendor/phpunit/phpunit/phpunit')) {
     define('PHPUNIT_COMPOSER_INSTALL', dirname(__DIR__).'/vendor/autoload.php');
     require PHPUNIT_COMPOSER_INSTALL;
-    PHPUnit\TextUI\Command::main();
+    exit((new \PHPUnit\TextUI\Application())->run($GLOBALS['argv']));
```
Comment (melroy89) [inline bin/phpunit:11]: This is needed when you upgrade to v10 phpunit. After all `PHPUnit\TextUI\Command` does not exists anymore.

**Comment by melroy89** [comment]
Ps. yes I updated PHPUnit to v10 (which is almost 9 months old now) and Symfony Bridge isn't updated after all this time. We don't depend on it, once PHPUnit bridge will finally support v10 we can run the composer update recipe. Our tests won't fail.

More info: https://github.com/symfony/symfony/issues/49069

File: config/kbin_routes/activity_pub.yaml (hunk 1 above)
Comment (e-five256) [reply config/kbin_routes/activity_pub.yaml:18]: `.well-known/nodeinfo`

seems to generate a 500 for me:

```
Unable to generate a URL for the named route "ap_node_info_v21" as such route does not exist.
```

Comment (melroy89) [reply config/kbin_routes/activity_pub.yaml:18]: WOH!? this is not good. Thanks for spotting.

Comment (melroy89) [reply config/kbin_routes/activity_pub.yaml:18]: I forgot about that. Fixed ;)

File: phpunit.xml.dist [hunk 3]
```diff
@@ -1,35 +1,24 @@
...
+    <server name="SHELL_VERBOSITY" value="-1"/>
+    <server name="SYMFONY_PHPUNIT_REMOVE" value=""/>
+    <server name="SYMFONY_PHPUNIT_VERSION" value="9.4"/>
+  </php>
+  <testsuites>
+    <testsuite name="Project Test Suite">
+      <directory>tests</directory>
+    </testsuite>
+  </testsuites>
+  <coverage/>
```
Comment (e-five256) [inline phpunit.xml.dist:15]: this is slightly strange to me that `include` and `exclude` were inside `coverage` and now it's not but I'm not sure I know enough to know why it changed

Comment (e-five256) [reply phpunit.xml.dist:15]: oh wait, no, this is right, include and exclude are members of source https://docs.phpunit.de/en/10.4/configuration.html#the-include-element

coverage has some others https://docs.phpunit.de/en/10.4/configuration.html#the-includeuncoveredfiles-attribute

File: src/Validator/Unique.php [hunk 4]
```diff
@@ -25,60 +22,31 @@ class Unique extends Constraint
...
+    ) {
+        parent::__construct([]);
 
         if (0 === \count($fields)) {
-            throw new InvalidOptionsException('fields option must have at least one field', ['fields']);
+            throw new InvalidOptionsException('`fields` option must have at least one field', ['fields']);
         }
 
-        if (!$options['entityClass']) {
+        if ('' === $this->entityClass) {
```
Comment (e-five256) [inline src/Validator/Unique.php:45]: I'm wondering if this also needs to do `null === $this->entityClass || '' === $this->entityClass)` or that gets checked elsewhere. I'm at least still able to hit unique validator errors so perhaps I'm mistaken, just noticed other things did when doing `'' ===`

Comment (melroy89) [reply src/Validator/Unique.php:45]: I agree

**Review by e-five256 — APPROVED** [review]

**Comment by melroy89** [comment]
Sorry.. I do now see issues:

```
{"message":"Uncaught PHP Exception TypeError: \"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33\" at /var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php line 1071","context":{"exception":{"class":"TypeError","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php:1071"}},"level":500,"level_name":"CRITICAL","channel":"request","datetime":"2023-11-17T03:38:12.698186+01:00","extra":{}}
{"message":"Exception thrown when handling an exception (Twig\\Error\\RuntimeError: Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33 at /var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig line 104)","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95"}}},"level":500,"level_name":"CRITICAL","channel":"request","datetime":"2023-11-17T03:38:12.704957+01:00","extra":{}}
{"message":"Uncaught Exception: Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"TypeError","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php:1071"}}}},"level":500,"level_name":"CRITICAL","channel":"php","datetime":"2023-11-17T03:38:12.705019+01:00","extra":{}}
{"message":"Uncaught PHP Exception Twig\\Error\\RuntimeError: \"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33\" at /var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig line 104","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"TypeError","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php:1071"}}}},"level":500,"level_name":"CRITICAL","channel":"request","datetime":"2023-11-17T03:38:12.705151+01:00","extra":{}}
{"message":"Exception thrown when handling an exception (Twig\\Error\\RuntimeError: Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33 at /var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig line 104)","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95"}}},"level":500,"level_name":"CRITICAL","channel":"request","datetime":"2023-11-17T03:38:12.708712+01:00","extra":{}}
{"message":"Uncaught Exception: Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"TypeError","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php:1071"}}}}}},"level":500,"level_name":"CRITICAL","channel":"php","datetime":"2023-11-17T03:38:12.708761+01:00","extra":{}}
```

Not sure what is happening. Maybe I need to split again the PR in smaller chunks, and test them in isolation. 

Maybe something changed again in Doctrine making it fail now.


```
{"message":"Uncaught Exception: Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","context":{"exception":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"Twig\\Error\\RuntimeError","message":"Error rendering \"sidebar_subscriptions\" component: Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/templates/layout/_sidebar.html.twig:104","previous":{"class":"Exception","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/symfony/ux-twig-component/src/Twig/ComponentExtension.php:95","previous":{"class":"TypeError","message":"Doctrine\\DBAL\\Connection::executeQuery(): Argument #1 ($sql) must be of type string, null given, called in /var/www/kbin.melroy.org/html/vendor/doctrine/orm/lib/Doctrine/ORM/Query/Exec/SingleSelectExecutor.php on line 33","code":0,"file":"/var/www/kbin.melroy.org/html/vendor/doctrine/dbal/src/Connection.php:1071"}}}}}},"level":500,"level_name":"CRITICAL","channel":"php","datetime":"2023-11-17T03:41:04.770302+01:00","extra":{}}
```

**Pull request closed without merging**